src/main/java/it/gov/pagopa/payment/service/payment/common/CommonAuthServiceImpl.java

```diff
@@ -81,13 +85,11 @@ protected AuthPaymentDTO invokeRuleEngine(String trxCode, TransactionInProgress
                 transactionInProgressRepository.updateTrxRejected(trx.getId(), authPaymentDTO.getRejectionReasons(), trx.getTrxChargeDate());
                 log.info("[TRX_STATUS][REJECTED] The transaction with trxId {} trxCode {}, has been rejected ",trx.getId(), trx.getTrxCode());
                 if (authPaymentDTO.getRejectionReasons().contains(RewardConstants.INITIATIVE_REJECTION_REASON_BUDGET_EXHAUSTED)) {
-                    throw new ClientExceptionWithBody(
-                            HttpStatus.FORBIDDEN,
+                    throw new BudgetExhaustedException(
```

Comment:
some Exception, like this, could have a default code that is supposed to be always the same: we could thus expose a constructor without CODE, using such default code in the constructor